proxy/http2/Http2ConnectionState.cc

```diff
@@ -1601,7 +1595,20 @@ Http2ConnectionState::send_goaway_frame(Http2StreamId id, Http2ErrorCode ec)
   SCOPED_MUTEX_LOCK(lock, this->ua_session->mutex, this_ethread());
   this->ua_session->handleEvent(HTTP2_SESSION_EVENT_XMIT, &frame);
 
-  handleEvent(HTTP2_SESSION_EVENT_FINI, nullptr);
+  this->ua_session->set_half_close_flag(true);
+  DebugHttp2Con(ua_session, "session half-close");
+
+  // As for gracefull shutdown, TS should process outstanding stream as long as possible.
+  // As for signal connection error, TS should close connection immediately.
+  if (ec != Http2ErrorCode::HTTP2_ERROR_NO_ERROR) {
```

Comment:
Between the two GOAWAYs, RFC mentions waiting for at least one RTT. It is the time that outstanding streams are processed. Otherwise, what's the point to send two GOAWAYs instead of one?